x-pack/filebeat/module/salesforce/login/ingest/pipeline.yml

```diff
@@ -0,0 +1,368 @@
+    ignore_missing: true
+- rename:
+    field: json.Application
+    target_field: salesforce.login.application
+    ignore_missing: true
+- rename:
+    field: json.PolicyId
+    target_field: salesforce.login.policy_id
+    ignore_missing: true
+- rename:
+    field: json.SessionLevel
+    target_field: salesforce.login.session_level
+    ignore_missing: true
+- rename:
+    field: json.EVENT_TYPE
+    target_field: salesforce.login.event_type
+    ignore_missing: true
+- rename:
+    field: json.REQUEST_ID
+    target_field: salesforce.login.request_id
+    ignore_missing: true
+- rename:
+    field: json.ORGANIZATION_ID
+    target_field: salesforce.login.organization_id
+    ignore_missing: true
+- rename:
+    field: json.RUN_TIME
+    target_field: salesforce.login.run_time
+    ignore_missing: true
+- rename:
+    field: json.CPU_TIME
+    target_field: salesforce.login.cpu_time
+    ignore_missing: true
+- rename:
+    field: json.DB_TOTAL_TIME
+    target_field: salesforce.login.db_total_time
+    ignore_missing: true
+- append:
+    field: related.ip
+    value: "{{{json.CLIENT_IP}}}"
+    ignore_failure: true
+    if: 'ctx?.json?.CLIENT_IP != "" && ctx?.json?.CLIENT_IP != null'
+- rename:
+    field: json.CLIENT_IP
+    target_field: salesforce.login.client_ip
+    ignore_missing: true
+- rename:
+    field: json.URI_ID_DERIVED
+    target_field: salesforce.login.uri_id_derived
+    ignore_missing: true
+
+#######################
+## ECS Event Mapping ##
+#######################
+
+- set:
+    field: event.outcome
+    value: success
+    if: 'ctx?.json?.LOGIN_STATUS == "LOGIN_NO_ERROR" && ctx?.json?.LOGIN_STATUS != null'
+    ignore_failure: true
+- set:
+    field: event.outcome
+    value: failure
+    if: 'ctx?.json?.LOGIN_STATUS != "LOGIN_NO_ERROR" && ctx?.json?.LOGIN_STATUS != null'
+    ignore_failure: true
+- set:
+    field: event.outcome
+    value: success
+    if: 'ctx?.json?.Status == "Success" && ctx?.json?.Status != null'
+    ignore_failure: true
+- set:
+    field: event.outcome
+    value: failure
+    if: 'ctx?.json?.Status != "Success" && ctx?.json?.Status != null'
+    ignore_failure: true
+- date:
+    field: json.CreatedDate
+    target_field: event.created
+    formats: 
+    - ISO8601
+    ignore_failure: true
+- rename:
+    field: json.LoginUrl
+    target_field: event.url
+    ignore_missing: true
+- set:
+    field: event.type
+    value: "info"
+- set:
+    field: event.kind
+    value: "event"
+- set:
+    field: event.action
+    value: "login-attempt"
+- set:
+    field: event.category
+    value: "authentication"
+- set:
+    field: event.dataset
+    value: "salesforce.login"
+- set:
+    field: event.module
+    value: "salesforce"
+
+######################
+## ECS User Mapping ##
+######################
+
+- rename:
+    field: json.USER_NAME
+    target_field: user.email
+    ignore_missing: true
+- rename:
+    field: json.Username
+    target_field: user.email
+    ignore_missing: true

```

Comment:
just double-checking: is username ALWAYS the email?

Reply:
As per [this article](https://help.salesforce.com/s/articleView?id=sf.basics_intro_usernames_passwords.htm&type=5), the username must be in the format of an email address, for example, jane@salesforce.com. The email used in your username need not function or match the email address used for the account. In addition to this, in all the live responses that we have seen so far, the username was an email address.

I hope this clarifies, please let me know if you think we should still change the mapping to `user.name`. Thanks!

Reply:
Thanks for checking this! I recommend leaving a comment in this file to preserve any confusion in the future.